src/drivers/imu/bmi088/BMI088_accel.cpp

```diff
@@ -0,0 +1,597 @@
+    _bad_transfers(perf_alloc(PC_COUNT, "bmi088_accel_bad_transfers")),
+    _bad_registers(perf_alloc(PC_COUNT, "bmi088_accel_bad_registers")),
+    _duplicates(perf_alloc(PC_COUNT, "bmi088_accel_duplicates")),
+    _got_duplicate(false)
+{
+    _px4_accel.set_device_type(DRV_ACC_DEVTYPE_BMI088);
+}
+
+BMI088_accel::~BMI088_accel()
+{
+    /* make sure we are truly inactive */
+    stop();
+
+    /* delete the perf counter */
+    perf_free(_sample_perf);
+    perf_free(_measure_interval);
+    perf_free(_bad_transfers);
+    perf_free(_bad_registers);
+    perf_free(_duplicates);
+}
+
+int
+BMI088_accel::init()
+{
+    /* do SPI init (and probe) first */
+    int ret = SPI::init();
+
+    /* if probe/setup failed, bail now */
+    if (ret != OK) {
+        DEVICE_DEBUG("SPI setup failed");
+        return ret;
+    }
+
+    return reset();
+}
+
+uint8_t
+BMI088_accel::read_reg(unsigned reg)
+{
+    // For the BMI088, you need to read out a dummy byte before you can read out the normal data (see section "SPI interface of accelerometer part" of the BMI088 datasheet)
+    uint8_t cmd[3] = { (uint8_t)(reg | DIR_READ), 0, 0};
+
+    transfer(cmd, cmd, sizeof(cmd));
+
+    return cmd[2]; // Skip dummy byte in cmd[1] and read out actual data
+}
+
+uint16_t
+BMI088_accel::read_reg16(unsigned reg)
+{
+    // For the BMI088, you need to read out the dummy byte before you can read out the normal data (see section "SPI interface of accelerometer part" of the BMI088 datasheet)
+    uint8_t cmd[4] = { (uint8_t)(reg | DIR_READ), 0, 0, 0 };
+
+    transfer(cmd, cmd, sizeof(cmd));
+
+    return (uint16_t)(cmd[2] << 8) | cmd[3]; // Skip dummy byte in cmd[1]
+}
+
+int BMI088_accel::reset()
+{
+    write_reg(BMI088_ACC_SOFTRESET, BMI088_SOFT_RESET); // Soft-reset
+    up_udelay(5000); // After a POR or soft-reset, the sensor needs up to 1ms boot time (see section "Power Modes: Accelerometer" in the BMI088 datasheet). Based off of testing it seems this value needs to be increased from 1ms. Setting it to 5ms.
+
+    // Perform a dummy read here to put the accelerometer part of the BMI088 back into SPI mode after the reset
+    // The dummy read basically pulls the chip select line low and then high
+    // See section "Serial Peripheral Interface (SPI)" of the BMI088 datasheet for more details.
+    read_reg(BMI088_ACC_CHIP_ID);
+
+    // Enable Accelerometer
+    // The accelerometer needs to be enabled first, before writing to its registers
+    write_checked_reg(BMI088_ACC_PWR_CTRL, BMI088_ACC_PWR_CTRL_EN);
+    up_udelay(5000); // After changing power modes, the sensor requires up to 5ms to settle. Any communication with the sensor during this time should be avoided (see section "Power Modes: Acceleromter" in the BMI datasheet)
+
+    // Set the PWR CONF to be active
+    write_checked_reg(BMI088_ACC_PWR_CONF, BMI088_ACC_PWR_CONF_ACTIVE); // Sets the accelerometer to active mode
+
+    // Write accel bandwidth and output data rate
+    // ToDo set the bandwidth
```

Comment:
I have not configured the bandwidth (ACC_CONF register). The actual on-chip filtering is a function of the output data rate as well (similarly to how it is on the gyro). (See section "4.4.1 Accelerometer" of the BMI088 datasheet).
@dagar Please advise as to how this should be configured.